File: UniversityGUI/src/Order.java
import java.time.LocalDate;
import java.util.ArrayList;

public class Order {
    private LocalDate date;
    private String orderNumber;
    private ArrayList<OrderLine> lines;

    public Order(String orderNumber) {
        this.orderNumber = orderNumber;
        this.lines = new ArrayList<OrderLines>();
    }

    public Order(String orderNumber, LocalDate date){
        this.orderNumber = orderNumber;
        this.date = date;
        this.lines = new Arraylist<>();
    }

    public double total(){
        //TODO
        double total= 0;
        for( OrderLine ol : lines) {
            total += ol.subTotal();
        }
        return total;
    }

    public String toString(){
        StringBuilder sb = new StringBuilder();
        sb.append("------------------------------------------------------------------------------------------------------------\n");
        sb.append(String.format("Order:[%-20s|%-10s]\n",this.getOrderNumber(),this.getDate()));
        for (OrderLine line: this.getLines()){
            sb.append(String.format("--->%-50s\n",line.toString()));
        }
        sb.append(String.format("---> Order Total: $%10.2f\n", this.total()));
        sb.append("============================================================================================================");
        return sb.toString();
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public String getOrderNumber() {
        return orderNumber;
    }

    public void setOrderNumber(String orderNumber) {
        this.orderNumber = orderNumber;
    }

    public ArrayList<OrderLine> getLines() {
        return lines;
    }

    public void setLines(ArrayList<OrderLine> lines) {
        this.lines = lines;
    }
}

File: PythonServer/FinalPythonShop/src/Order.py
#import java.time.LocalDate;
#import java.util.ArrayList;
import OrderLine

class Order:
    #private LocalDate date;
    #private String orderNumber;
    #private ArrayList<OrderLine> lines;

    def __init__(self, orderNumber:str, date=None):
        self.orderNumber = orderNumber
        self.date = date
        self.lines = list()

    def total(self):
        #//TODO
        total= 0;
        for ol in self.lines:
            total += ol.subTotal()
        return total

    def toString(self):
        #todo: make sure this works
        sb = ""
        sb += "------------------------------------------------------------------------------------------------------------\n"
        sb += str("Order:[%-20s|%-10s]\n",self.getOrderNumber(),self.getDate())
        for line in self.lines:
            sb+= str("--->%-50s\n",line.toString())
        sb += str("---> Order Total: $%10.2f\n", self.total())
        sb += "============================================================================================================"
        return sb

    def getDate(self):
        return self.date

    #todo:update LocalDate
    def setDate(self, date:LocalDate):
        self.date = date

    def getOrderNumber(self):
        return self.orderNumber

    def setOrderNumber(self, orderNumber:str):
        self.orderNumber = orderNumber

    def getLines(self):
        return self.lines

    def setLines(self, lines:list):
        self.lines = lines